scripts/attendance.py

```diff
@@ -0,0 +1,167 @@
+DAILY_WORKING_HOURS = 9
+
+
+class AttendanceManagement():
+
+    def __init__(self):
+        self.today = datetime.now()
+        self.current_day = self.today.day
+        self.current_month = self.today.month
+        self.current_year = self.today.year
+
+    def time_to_minutes(self, time):
+        """
+        This accepts time format in put and convert it into minutes in integer
+         format
+        :param time:
+        :return: minutes in integer format
+        """
+
+        str_time = str(time)
+        time_list = str_time.split(":")
+        hours, minutes = int(time_list[0]), int(time_list[1])
+        ret_minutes = (hours * 60) + minutes
+        return ret_minutes
+
+    def get_saturday_sunday_count(self, days):
+        """
+        This accepts days liost from current month and return total number of
+         saturaday and sunday count.
+        :param days:
+        :return: saturady_sunday_count
+        """
+
+        saturady_sunday_count = 0
+        for day in days:
+            if day[5] != 0:
+                saturady_sunday_count += 1
+            if day[6] != 0:
+                saturady_sunday_count += 1
+        return saturady_sunday_count
+
+    def get_total_working_days_remaining(self, curent_month, current_year,
+                                         present_days,
+                                         number_of_holiday_remaining=0):
+        """
+        This will return total workinga days in current month after
+        removing holiday and saturday-sunday from month days.
+        :param curent_month:
+        :param current_year:
+        :param present_days:
+        :param number_of_holiday_remaining:
+        :return: total_working_day_remaining
+        """
+
+        total_days_in_month = monthlen(current_year, curent_month)
+        days = Calendar().monthdayscalendar(current_year, curent_month)
+        saturady_sunday_count = self.get_saturday_sunday_count(days)
+        saturady_sunday_count += number_of_holiday_remaining
+        total_working_day_remaining = \
+            (total_days_in_month - (saturady_sunday_count + present_days))
+        return total_working_day_remaining
+
+    def get_avarage_working_target(self, average_attendace,
+                                   available_working_days, present_days):
+        """
+        This method do actual attendance calculation.
+
+        :param average_attendace:
+        :param available_working_days:
+        :param present_days:
+        :return:
+        """
+        if average_attendace == '09:00:00' or\
+            average_attendace.startswith('09:00:') or\
+                average_attendace.startswith('9:00:'):
+            return 9, 00, available_working_days
+        avrage_working_time_list = average_attendace.split(":")
+        average_attendace_hours = int(avrage_working_time_list[0])
+        average_attendace_minutes = int(avrage_working_time_list[1])
+        if average_attendace_hours < DAILY_WORKING_HOURS:
```

Comment:
We need to add comments in the code, which will help to understand the code better